### PR TITLE
Make it possible to keep comments

### PR DESCRIPTION
**Pull request by satabin**
Comments can be useful to keep when transforming XML from source to
source.

### CODE REVIEW

File: xml/src/main/scala/fs2/data/xml/package.scala [hunk 1]
```diff
@@ -29,11 +29,24 @@ package object xml {
...
+  /** Transforms a stream of characters into a stream of XML events.
+    * Emitted tokens are guaranteed to be valid up to that point.
+    * If the streams ends without failure, the sequence of tokens is ensured
+    * to represent a (potentially empty) sequence of valid XML documents.
+    *
+    * if `skipComment` is `false`, then comment event will be emitted
+    * together with the comment content.
+    */
+  def events[F[_], T](
+      skipComment: Boolean = true)(implicit F: RaiseThrowable[F], T: CharLikeChunks[F, T]): Pipe[F, T, XmlEvent] =
```
Comment (ybasket) [inline xml/src/main/scala/fs2/data/xml/package.scala:48]: At least my mental model prefers something like `includeComments` with a default of `false`, wdyt?

**Review by ybasket — APPROVED** [review]
One comment on naming, apart from that it looks good!

File: xml/src/main/scala/fs2/data/xml/package.scala (hunk 1 above)
Comment (satabin) [reply xml/src/main/scala/fs2/data/xml/package.scala:48]: I was thinking in "true by default", but sure.

Comment (ybasket) [reply xml/src/main/scala/fs2/data/xml/package.scala:48]: I come from "opt-in to extra work" and if true, then comment.isDefined (pseudo) also being true.

Both views make sense to me, so merge as you think is best.

Comment (satabin) [reply xml/src/main/scala/fs2/data/xml/package.scala]: I made it use less negation, it probably improves code readablity, thanks.

**Pull request merged**